experiments/process_sandbox/include/process_sandbox/sandbox.h

```diff
@@ -20,95 +21,383 @@
 
 #include <snmalloc.h>
 
-#ifndef SANDBOX_PAGEMAP
-#  ifdef SNMALLOC_DEFAULT_PAGEMAP
-#    define SANDBOX_PAGEMAP SNMALLOC_DEFAULT_PAGEMAP
-#  else
-#    define SANDBOX_PAGEMAP snmalloc::SuperslabMap
-#  endif
-#endif
-
-namespace snmalloc
-{
-  template<class T>
-  class MemoryProviderStateMixin;
-  template<class T>
-  class PALPlainMixin;
-  template<
-    bool (*NeedsInitialisation)(void*),
-    void* (*InitThreadAllocator)(function_ref<void*(void*)>),
-    class MemoryProvider,
-    class ChunkMap,
-    bool IsQueueInline>
-  class Allocator;
-  template<typename T>
-  struct SuperslabMap;
-  void* no_replacement(void*);
-}
-
 namespace sandbox
 {
   struct SharedMemoryRegion;
-  struct SharedPagemapAdaptor;
-  struct MemoryProviderBumpPointerState;
   class CallbackDispatcher;
   class ExportedFileTree;
   struct CallbackHandlerBase;
+  class Library;
 
   /**
    * An snmalloc Platform Abstraction Layer (PAL) that cannot be used to
    * allocate memory.  This is used for sandboxes, where all memory comes from a
    * pre-defined shared region.
    */
   using NoOpPal = snmalloc::PALNoAlloc<snmalloc::DefaultPal>;
+
   using snmalloc::pointer_offset;
 
   /**
-   * The memory provider for the shared region.  This manages a single
-   * contiguous address range.
+   * Snmalloc back-end structure for shared memory allocations.  This defines
+   * how snmalloc will interact with allocations that are per-sandbox.
+   *
+   * Globals used by the snmalloc instances that allocate sandbox memory from
+   * the outside.
    */
-  class SharedMemoryProvider
-  : public snmalloc::MemoryProviderStateMixin<NoOpPal>
+  struct SharedAllocConfig : public snmalloc::CommonConfig
   {
     /**
-     * Base address of the shared memory region.
+     * Forward definition of `Pagemap` so that `LocalState` can refer to it.
```

Comment:
Is this still the case, since I cannot see the reference in the code, but I might be missing it?

Reply:
Yup, dead code.  Thanks!